Cython/Distutils/build_ext.py

```diff
@@ -5,8 +5,8 @@
     try:
         from setuptools.command.build_ext import build_ext as _build_ext
     except ImportError:
-        # We may be in the process of importing setuptools, which tries
-        # to import this.
+        # Since Cython.Distutils.build_ext is being imported in setuptools,
+        # we may be in the process of importing setuptools.
         from distutils.command.build_ext import build_ext as _build_ext
```

Comment:
Ok, but if setuptools is trying to make use of Cython, which then makes Cython use distutils instead, that's really going in the wrong direction. Not sure how we can work around that, but this isn't a good way.

Reply:
Looking at what setuptools does:
https://github.com/pypa/setuptools/blob/9f1822ee910df3df930a98ab99f66d18bb70659b/setuptools/command/build_ext.py#L16
It actually uses the `build_ext` from distutils, unless there is one provided by Cython – which inherits from the one in distutils.

Meaning, this whole attempt to use something from setuptools is wrong. Cython shouldn't try to inherit from itself. We should always be using `build_ext` from distutils to inherit from, never the one from setuptools, so that setuptools can decide which of the two to choose.

Reply:
Yes it make sense.